kola/harness.go

```diff
@@ -432,10 +432,25 @@ func getClusterSemver(flight platform.Flight, outputDir string) (*semver.Version
 
 	out, stderr, err := m.SSH("grep ^VERSION_ID= /etc/os-release")
 	if err != nil {
-		return nil, fmt.Errorf("parsing /etc/os-release: %v: %s", err, stderr)
+		return nil, fmt.Errorf("parsing /etc/os-release for VERSION_ID: %v: %s", err, stderr)
 	}
 	ver := strings.Split(string(out), "=")[1]
 
+	out, stderr, err = m.SSH("grep ^BUILD_ID= /etc/os-release | { grep -o 'dev-.*-nightly' || true ; } | cut -d - -f 2-  | rev | cut -d - -f 2- | rev")
+	if err != nil {
+		return nil, fmt.Errorf("parsing /etc/os-release for nightly branch in BUILD_ID: %v: %s", err, stderr)
+	}
+	branch := string(out)
+	if branch == "main" || branch == "flatcar-master" {
+		// "main" is a nightly build of the main branch,
+		// "flatcar-master" refers to the manifest branch where dev builds are started
+		ver = "9999.99.99"
```

Comment:
I'd make it `999999.99.99`. We don't need to saddle ourselves with "year 2040" problem, right after "year 2038" one. ;)

Reply:
Yes, let's trade it for a "year 2286" problem

Reply:
Wait, you had one 9 more, "year 4752" - even better!

Reply:
My reasoning was to stay < 32767 in case someone swaps out the semver lib to one that is more limited